client/frontend/src/components/server/Console.vue

```diff
@@ -74,39 +74,53 @@ function clearConsole() {
   if (console.value) console.value.replaceChildren([])
 }
 
-const history = []
-let historyIndex = -1
+const history = ref([])
+const historyIndex = ref(-1)
+const temporaryCommand = ref('')
 
 function sendCommand() {
-  if (command.value) {
-    history.push(command.value)
-    historyIndex = -1
+  if (historyIndex.value !== -1) {
+    history.value.splice(historyIndex.value, 1)
+  }
```

Comment:
while this takes care of history access duplicating history entries you'd still get duplicates by manually typing the same command again, when done in direct succession that still causes the up and down keys to not produce a noticeable change
when a command is entered without using the history, please at least check that the most recent command in history isn't the same as one entered to ensure that going through the history always has noticeable changes when not at the end of the history